src/main/java/duke/CommandParser.java

```diff
@@ -0,0 +1,150 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.action.Action;
+import duke.action.Delete;
+import duke.action.Echo;
+import duke.action.Farewell;
+import duke.action.Mark;
+import duke.action.Match;
+import duke.action.MyList;
+import duke.action.TaskList;
+import duke.action.Unmark;
+import duke.exception.EmptyDescriptionException;
+import duke.exception.NoIndexException;
+import duke.exception.UnknownCommandException;
+import duke.exception.WrongDateFormatException;
+import duke.exception.WrongDateOrderingException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * Parses user commands and performs corresponding actions in the Duke application.
+ */
+
+public class CommandParser {
+
+    /**
+     * Parses user commands and performs corresponding actions in the Duke application.
+     */
+    public static Action parseCommand(String command, TaskList taskList) throws
+            EmptyDescriptionException, NoIndexException, UnknownCommandException,
+            WrongDateFormatException, WrongDateOrderingException {
+        try {
+            String[] words = command.split(" ");
+
+            switch (words[0]) {
+            case "find":
+                if (words.length > 1) {
+                    String keyword = command.substring(5).trim();
+                    taskList.matches(keyword);
+                } else {
+                    System.out.println("Please provide a keyword to search for.");
```

Comment:
In most other instances of commands being used, you throw some kind of exception if the input format is not of the intended form. Is there a reason that you print to the console in this case?